src/main/java/org/graylog/integrations/aws/service/KinesisService.java

```diff
@@ -380,4 +387,42 @@ Record selectRandomRecord(List<Record> recordsList) {
         LOG.debug("Selecting a random Record from the sample list.");
         return recordsList.get(new Random().nextInt(recordsList.size()));
     }
+
+    /**
+     * Creates a new Kinesis stream.
+     *
+     * @param kinesisNewStreamRequest request which contains region, access, secret, region, streamName and shardCount
+     * @return the status response
+     */
+    public KinesisNewStreamResponse createNewKinesisStream(KinesisNewStreamRequest kinesisNewStreamRequest) {
+        LOG.debug("Creating Kinesis client with the provided credentials.");
+        KinesisClient kinesisClient = createClient(kinesisNewStreamRequest.region(),
+                                                   kinesisNewStreamRequest.awsAccessKeyId(),
+                                                   kinesisNewStreamRequest.awsSecretAccessKey());
+
+        LOG.debug("Creating new Kinesis stream request [{}].", kinesisNewStreamRequest.streamName());
+        CreateStreamRequest createStreamRequest = CreateStreamRequest.builder()
+                                                                     .streamName(kinesisNewStreamRequest.streamName())
+                                                                     .shardCount(SHARD_COUNT)
+                                                                     .build();
+        LOG.debug("Sending request to create new Kinesis stream [{}] with [{}] shards.",
+                  kinesisNewStreamRequest.streamName(), SHARD_COUNT);
+
+        String responseMessage;
```

Comment:
Although this `responseMessage` variable works as defined now, is not actually needed outside of the try/catch block, so it can be declared in each independently. This ensures that it's unavailable outside the scope where it's needed. Something like the following example. What do you think?

```
try {
    kinesisClient.createStream(createStreamRequest);
    final String responseMessage = String.format("Success. The new stream [%s] was created with [%d] shards.",
                                           kinesisNewStreamRequest.streamName(), SHARD_COUNT);
    return KinesisNewStreamResponse.create(responseMessage, new HashMap<>());
} catch (Exception e) {
    final String specificError = ExceptionUtils.formatMessageCause(e);
    final String responseMessage = String.format("Attempt to create [%s] new Kinesis stream" +
```

Reply:
Good idea!